### PR TITLE
CR-14554-ir

### PR DESCRIPTION
**Pull request by yaroslav-codefresh**

### CODE REVIEW

File: cmd/commands/runtime.go [hunk 1]
```diff
@@ -916,6 +933,55 @@ func NewRuntimeLogsCommand() *cobra.Command {
 	return cmd
 }
 
+func migrateInternalRouter(ctx context.Context, opts *RuntimeUpgradeOptions, newRt *runtime.Runtime) error {
+	dbRuntime, err := cfConfig.NewClient().V2().Runtime().Get(ctx, opts.RuntimeName)
```
Comment (ghost) [inline cmd/commands/runtime.go]: just call `getRuntime` (def is in `common.go`)

File: cmd/commands/runtime_install.go [hunk 2]
```diff
@@ -20,10 +20,13 @@ import (
 	"encoding/hex"
 	"errors"
 	"fmt"
+	billyUtils "github.com/go-git/go-billy/v5/util"
```
Comment (ghost) [inline cmd/commands/runtime_install.go:23]: it doesn't matter much - but why did the imports jump around like this?

File: cmd/commands/runtime_install.go [hunk 3]
```diff
@@ -1036,10 +1053,33 @@ func installComponents(ctx context.Context, opts *RuntimeInstallOptions, rt *run
...
+					useGatewayAPI:             opts.useGatewayAPI,
+				}, rt)
 			},
 		}); err != nil {
-			return fmt.Errorf("failed to patch Argo-Workflows ingress: %w", err)
+			return fmt.Errorf("failed to patch Internal Router ingress: %w", err)
+		}
+	}
+
+	if opts.shouldInstallIngress() && rt.Spec.IngressController != string(routingutil.IngressControllerALB) {
```
Comment (ghost) [inline cmd/commands/runtime_install.go]: isn't that the exact same condition as line 1053? can't it be inside the same `if`?

File: manifests/default-resources/internal-router/kustomization.yaml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+bases:
+  - ../../internal-router
```
Comment (ghost) [inline manifests/default-resources/internal-router/kustomization.yaml:4]: is this file being used anywhere?

Comment (yaroslav-codefresh) [reply manifests/default-resources/internal-router/kustomization.yaml:4]: not sure I even understand the purpose of the `default-resources` folder -- just did it the same as for other components

**Review by noam-codefresh — DISMISSED** [review]

**Comment by noam-codefresh** [comment]
i am working on quickly fixing the upgrade regression as well (in this pr or in another), and then i'll re-approve this one.
anyway - be safe.

**Review by noam-codefresh — APPROVED** [review]

**Pull request merged**